src/MLOps.NET/Catalogs/EvaluationCatalog.cs

```diff
@@ -49,7 +49,10 @@ public async Task LogMetricAsync(Guid runId, string metricName, double metricVal
 
             foreach (var metric in properties)
             {
-                await LogMetricAsync(runId, metric.Name, (double)metric.GetValue(metrics));
+                if (!(double.IsNaN((double)metric.GetValue(metrics))))
+                {
+                    await LogMetricAsync(runId, metric.Name, (double)metric.GetValue(metrics));
```

Comment:
```suggestion
                    await LogMetricAsync(runId, metric.Name, metric);
```

Reply:
Should this - await LogMetricAsync(runId, metric.Name, metric);

not be this - await LogMetricAsync(runId, metric.Name, value);

Reply:
Totally correct, updated my suggestion.